src/foam/u2/wizard/StepWizardController.js

```diff
@@ -91,8 +91,13 @@ foam.CLASS({
     {
       name: 'currentSection',
       expression: function(currentWizardlet, wizardPosition$sectionIndex) {
-        return this.currentWizardlet.currentSection = currentWizardlet
-            .sections[wizardPosition$sectionIndex];
+        var maybeCurrentSectionFound = currentWizardlet.sections[wizardPosition$sectionIndex];
+
+        if ( maybeCurrentSectionFound ){
+          this.currentWizardlet.currentSection = maybeCurrentSectionFound;
+        }
+
+        return this.currentWizardlet.currentSection;
```

Comment:
@tharmaman does this work?

```
return currentWizardlet.sections[wizardPosition$sectionIndex];
```

Reply:
@tharmaman it's a synchronization issue. Currently testing a better fix for this

Reply:
```
    {
      name: 'currentSection',
      expression: function(
        wizardlets,
        wizardPosition$wizardletIndex, wizardPosition$sectionIndex
      ) {
        return wizardlets[wizardPosition$wizardletIndex]
          .sections[wizardPosition$sectionIndex];
      }
    },
```